### ISSUE TITLE
Warning 'missing semicolon' is a bit buggy

### ISSUE THREAD

**Issue by PalleZingmark** [rossPatton/stylint#211]
This behaviour gave me some new gray hairs yesterday when exploring some irregular warnings. 
We´re using `v1.2.3`. 

**test 1:**
``` css
.barfoo {
  background: red;
}
.foobarfo {
  &:before,
  &:after {
    background: red;
  }
}
.bar {
  background: red;
}
```
`-> Warning: missing semicolon, Line: 5: &:before`

**test 2:**
```css
.barfoo {
  background: red;
}
.foobarf {
  &:before,
  &:after {
    background: red;
  }
}
.bar {
  background: red;
}
```
`-> No warning`

**test 3:**
```css
.foobarfo {
  &:before,
  &:after {
    background: red;
  }
}
.bar {
  background: red;
} 
```
`-> No warning`

**test 4:**
```css 
.bar {
  background: red;
}
.foobarfoo {
  &:before,
  &:after {
    background: red;
  }
}
```
`-> Warning: missing semicolon, Line: 5: &:before`

**test 5:**
```css
.bar {
  background: red;
}
.foobarf {
  &:before,
  &:after {
    background: red;
  }
} 
```
`-> No warning`

As you can see, it's not only where in the context the code is, it's also relevant to the length of the classname it self.

**Comment by PalleZingmark** [comment]
**test 6:**

```css
.svg {
  background: $background;
  path,
  rect,
  polygon,
  circle {
    fill: $foreground;
  }
}
``` 
`-> Warning: missing semicolon, Line: 4: rect`

**Comment by dennishall1** [comment]
Checking out version 1.2.5 .. seems to say error: missing semicolon for *every* BEM-style classname in a group:

````
.module-class
    &__anything
        padding: 20px;
````

"missing semicolon" on line 2

**Comment by rossPatton** [comment]
this should be fixed in 1.3.0
